extension/src/paymentHandler/web-component-service.js

```diff
@@ -0,0 +1,50 @@
+const fetch = require('node-fetch')
+const { serializeError } = require('serialize-error')
+
+const configLoader = require('../config/config')
+const config = configLoader.load()
+
+function getOriginKeys (getOriginKeysRequest) {
+  return callAdyen('originKeys', getOriginKeysRequest)
+}
+
+function getPaymentMethods (getPaymentMethodsRequest) {
+  return callAdyen('paymentMethods', getPaymentMethodsRequest)
+}
+
+async function callAdyen(endpoint, requestString) {
+  let request = buildRequest(requestString)
+  let response
+  try {
+    response = await fetchAsync(endpoint, request)
+  } catch (err) {
+    response = serializeError(err)
+  }
+  return { request, response }
+}
+
+function buildRequest (requestString) {
+  const body = JSON.stringify({
+    merchantAccount: config.adyen.merchantAccount,
```

Comment:
What if merchantAccount is already supplied within payload?

Reply:
This is a good example, and yes possible, I've also a scenario that the requestString would be not a valid JSON, let me think what I could design a better structure for those... I could not find a better check yet, and this is a todo on a test. I will try to address them.

Reply:
https://github.com/commercetools/commercetools-adyen-integration/blob/231-add-payment-methods-and-origin-keys/extension/src/web-component-service.js#L34-L39

Reply:
additionally removing additional data from the response is moved here from payment-utils.